src/plugins/unified_histogram/public/chart/histogram.tsx

```diff
@@ -203,6 +203,22 @@ export function Histogram({
       transform: translate(-50%, -50%);
     }
   `;
+  // console.log('histogram', { lensProps, requestData, request, dataView, visContext });
+  if (
+    dataView.id &&
+    (dataView.id !== visContext.requestData.dataViewId ||
+      (!dataView.isPersisted() && !lensProps.attributes.state.adHocDataViews?.[dataView.id]))
+  ) {
+    // to prevent a race condition when the data view changes id
+    // the ids of the data view and the lens props and visContextDataView should be the same
+    // console.log('flakydebugging', {
+    //  dataViewId: dataView.id,
+    //  visContextDataViewId: visContext.requestData.dataViewId,
+    //  lensProps: lensProps.attributes.references.find((r) => r.id === dataView.id),
+    // });
+
+    return <></>;
+  }
```

Comment:
So, this seems to prevent the history to fail, because we have different data views in different props that are passed into the Lens embeddable. Ideally this should be caught on a higer level, so this kind of situation can't happen. 
However, if this resolves the race condition, it's now clear what's causing it.

Reply:
Wow! Thx! This is great news!

Reply:
Also, another update to tests would be necessary once you unskip them as in https://github.com/elastic/kibana/pull/200583/commits/e9d5c259e8134c2755c3e82d7c2b4d8891ccea8e

Reply:
tests are green, and flaky test runner too